modules/core/src/test/scala/org/tessellation/infrastructure/rewards/RewardsSuite.scala

```diff
@@ -0,0 +1,135 @@
+      SecurityProvider.forAsync[IO],
+      Resource.liftK(Random.scalaUtilRandom[IO])
+    ).mapN((_, _, _))
+
+  implicit val order: Ordering[Id] = (x: Id, y: Id) => 1
+
+  val config = RewardsConfig.default
+
+  val softStaking = SoftStaking.make(config.softStaking)
+  val dtm = DTM.make(config.dtm, 1.minutes)
+  val stardust = StardustCollective.make(config.stardust)
+  val hash = Hash("")
+
+  val facilitators = NonEmptySet.fromSetUnsafe(
+    SortedSet(
+      Id(
+        Hex(
+          "f2b230bb6ca3db890e647c171adbaeea27fc0191b7d84790c020c7fe0e04003c30a62d475af1300d8205a2436d640a1a091b7d53feb5a11bbb000013f75bdcf9"
+        )
+      ),
+      Id(
+        Hex(
+          "c7f9a08bdea7ff5f51c8af16e223a1d751bac9c541125d9aef5658e9b7597aee8cba374119ebe83fb9edd8c0b4654af273f2d052e2d7dd5c6160b6d6c284a17c"
+        )
+      )
+    )
+  )
+
+  test("epoch 1 snapshot reward is valid") { res =>
+    implicit val (_, sp, rand) = res
+
+    for {
+      rewards <- Rewards.make[IO](config, 1.minutes, softStaking, dtm, stardust).pure[F]
+      txs <- rewards
+        .calculateRewards(
+          EpochProgress(1L),
+          facilitators
+        )
+      sum = txs.toList.map(_.amount.value.toLong).sum
+
+    } yield expect(sum == 65843621389L)
+  }
+
+  test("epoch 2 snapshot reward is valid") { res =>
+    implicit val (_, sp, rand) = res
+
+    for {
+      rewards <- Rewards.make[IO](config, 1.minutes, softStaking, dtm, stardust).pure[F]
+      txs <- rewards
+        .calculateRewards(
+          EpochProgress(1296001L),
+          facilitators
+        )
+      sum = txs.toList.map(_.amount.value.toLong).sum
+
+    } yield expect(sum == 32921810694L)
+  }
+
+  test("epoch 3 snapshot reward is valid") { res =>
+    implicit val (_, sp, rand) = res
+
+    for {
+      rewards <- Rewards.make[IO](config, 1.minutes, softStaking, dtm, stardust).pure[F]
+      txs <- rewards
+        .calculateRewards(
+          EpochProgress(1296001L * 2),
+          facilitators
+        )
+      sum = txs.toList.map(_.amount.value.toLong).sum
+
+    } yield expect(sum == 16460905347L)
+  }
+
+  test("epoch 4 snapshot reward is valid") { res =>
+    implicit val (_, sp, rand) = res
+
+    for {
+      rewards <- Rewards.make[IO](config, 1.minutes, softStaking, dtm, stardust).pure[F]
+      txs <- rewards
+        .calculateRewards(
+          EpochProgress(1296001L * 3),
+          facilitators
+        )
+      sum = txs.toList.map(_.amount.value.toLong).sum
+
+    } yield expect(sum == 8230452674L)
+  }
+
+  test("epoch 4+n snapshot reward is 0") { res =>
+    implicit val (_, sp, rand) = res
+
+    for {
+      rewards <- Rewards.make[IO](config, 1.minutes, softStaking, dtm, stardust).pure[F]
+      txs <- rewards
+        .calculateRewards(
+          EpochProgress(1296001L * 4),
+          facilitators
+        )
+      sum = txs.toList.map(_.amount.value.toLong).sum
+
+    } yield expect(sum == 0)
+  }
```

Comment:
I think that there should be a test that checks if:

The sum of rewards in `RewardsConfig.rewardsPerEpoch` sums up the expected value which is `160000000000000000L`.

It might be the case that it won't (I haven't checked it), but in that case we have 2 options:
* change the value from `160000000000000000L` to one that's possible to distribute having the current length of the epoch and a constraint that each next epoch has exactly half of the rewards of the previous epoch (this allows us also to write a test that verifies this constraint)
* introduce a short 'reminder epoch' at the end of each epoch (or at the end of all epochs), where we distribute missing dags, so that they all sum up to the `160000000000000000L`

I think that the 1st option is better since it simplifies stuff (and this is how probably it works currently in the mainnet), but it has to be probably communicated to the CL management.

Reply:
Yeah. We were discussing the need of keeping total pool in snapshot but imo we should just keep that value in a suite and write another test to not make the Snapshot bigger (I know, it is just a Long value but still, the less we keep in snapshot the better)

I would go for 1st option.

Reply:
>We were discussing the need of keeping total pool in snapshot 

I think we were discussing keeping the total amount of rewards minted so far in the snapshot. Copying a value ~`160000000000000000L` from snapshot to snapshot wouldn't make much sense.

Reply:
Right. It was about already minted value.